internal/transport/http2_client.go

```diff
@@ -1215,7 +1217,7 @@ func (t *http2Client) handleGoAway(f *http2.GoAwayFrame) {
 	default:
 		t.setGoAwayReason(f)
 		close(t.goAway)
-		t.controlBuf.put(&incomingGoAway{})
+		defer t.controlBuf.put(&incomingGoAway{})
```

Comment:
Please add a comment explaining that we defer because `t.mu` is held

Reply:
Added "Defer as t.mu is currently held".